server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -379,6 +414,31 @@ public KeywordFieldMapper build(MapperBuilderContext context) {
                 this
             );
         }
+
+        private FieldType resolveFieldType(final IndexSortConfig indexSortConfig, final IndexMode indexMode, final String fullFieldName) {
+            if (shouldUseDocValuesSparseIndex(indexSortConfig, indexMode, fullFieldName)) {
+                return new FieldType(Defaults.FIELD_TYPE_WITH_SKIP_DOC_VALUES);
+            }
+            return new FieldType(Defaults.FIELD_TYPE);
+        }
+
+        private boolean shouldUseDocValuesSparseIndex(
+            final IndexSortConfig indexSortConfig,
+            final IndexMode indexMode,
+            final String fullFieldName
+        ) {
+            if (FieldMapper.DOC_VALUES_SPARSE_INDEX.isEnabled() == false) {
+                return false;
+            }
+            boolean isIndexedAndDocValuesDefault = indexed.isConfigured() || hasDocValues.isConfigured();
+            boolean isNotIndexedAndHasDocValues = indexed.getValue() || hasDocValues.getValue() == false;
+            boolean isLogsDbMode = IndexMode.LOGSDB.equals(indexMode);
+            boolean isHostNameField = HOST_NAME.equals(fullFieldName);
+            boolean isPrimarySortField = indexSortConfig != null && indexSortConfig.hasPrimarySortOnField(HOST_NAME);
+
+            return (isIndexedAndDocValuesDefault || isNotIndexedAndHasDocValues) && isLogsDbMode && isHostNameField && isPrimarySortField;
```

Comment:
I think we want to enable sparse index only if index has not been configured and doc values isn't disabled.

So I think this is easier:

```
indexed.isConfigured() == false && hasDocValues.getValue() == false && isLogsDbMode && isHostNameField && isPrimarySortField;
``` 

?

Reply:
`hasDOcValues.getValue() == false`? We need doc values to create the sparse index...